File: VB/CustomWizardExample/Wizard/CustomReportWizardPageFactory.vb
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.DataAccess.Wizard.Views
Imports DevExpress.XtraReports.Wizards3

Namespace CustomWizardExample.Wizard
	Public Class CustomReportWizardPageFactory
		Inherits XtraReportWizardPageFactory(Of Model.CustomReportModel, IDataSourceWizardClient)

		Public Sub New(ByVal client As IDataSourceWizardClient)
			MyBase.New(client)
		End Sub

		Protected Overrides Sub RegisterDependencies()
			MyBase.RegisterDependencies()

			Container.RegisterType(Of ChooseDataSourceTypePage(Of Model.CustomReportModel, IDataSourceWizardClient), Presenters.ChooseDataSourceTypePageEx)()

			Container.RegisterType(Of IChooseDataSourceTypePageView, Views.ChooseDataSourceTypePageViewEx)()

			Container.RegisterType(Of Presenters.ChoosePersonsDataSourceTypePage)()
			Container.RegisterType(Of Views.IChoosePersonsDataSourcePageView, Views.ChoosePersonsDataSourcePageView)()
		End Sub

	End Class
End Namespace

File: CS/CustomWizardExample/Wizard/CustomReportWizardPageFactory.cs
using DevExpress.DataAccess.Wizard;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.DataAccess.Wizard.Views;
using DevExpress.XtraReports.Wizards3;

namespace CustomWizardExample.Wizard {
    public class CustomReportWizardPageFactory : XtraReportWizardPageFactory<Model.CustomReportModel, IDataSourceWizardClient> {
        public CustomReportWizardPageFactory(IDataSourceWizardClient client) : base(client) { }

        protected override void RegisterDependencies() {
            base.RegisterDependencies();

            Container.RegisterType<ChooseDataSourceTypePage<Model.CustomReportModel, IDataSourceWizardClient>, Presenters.ChooseDataSourceTypePageEx>();

            Container.RegisterType<IChooseDataSourceTypePageView, Views.ChooseDataSourceTypePageViewEx>();

            Container.RegisterType<Presenters.ChoosePersonsDataSourceTypePage>();
            Container.RegisterType<Views.IChoosePersonsDataSourcePageView, Views.ChoosePersonsDataSourcePageView>();
        }

    }
}
